modules/security/pages/gsql-file-input-policy.adoc

```diff
@@ -0,0 +1,430 @@
+====
+
+==== RUN LOADING JOB
+
+When starting to run a file loading job check the given path and apply the file input policy to it.
+
+Below are two example cases:
+
+. Path for local data file:
+
++
+[console, gsql]
+----
+CREATE LOADING JOB load_ip FOR GRAPH test_graph {
+    DEFINE FILENAME f1 = "any:/tmp/data1.csv";;
+    LOAD f1
+        TO VERTEX account VALUES ($0, $0),
+        TO VERTEX actorIP VALUES ($1, $1),
+        TO EDGE event_property VALUES ($0 account, $1 actorIP)
+        ;
+}
+
+// Notice the path can be either absolute or relative
+RUN LOADING JOB load_ip USING f1="m1:./resources/data_set/gsql/k_step_neighber.csv"
+----
+
+. Path for config file:
++
+[console, gsql]
+----
+CREATE DATA_SOURCE KAFKA ka = "/tmp/kafka_broker.json" FOR GRAPH g
+CREATE LOADING JOB load_kafka {
+    // This path should also be considered
+    DEFINE FILENAME f1 = "$ka:/tmp/kafka_topic.json";
+    LOAD f1 TO VERTEX v1 VALUES($0, $1);
+}
+----
+
+==== Loading job from Directory
+
+Additionally, a loading job can use a directory instead of specific data file path for `FILENAME`.
+For example, when using the directory `/dir_1` as `FILENAME`, the TigerGraph loader will traverse all files in the directory to load data.
+
+If users define the input policy as `['/dir_1', '!/dir_1/data_1']`, so that the directory `/dir_1` is in `allowlist`, while a file `/dir_1/data_1` is in `blocklist` the TigerGraph Loader will skip the data file `/dir_1/data_1`.
+While still loading other files that are not in `blocklist`. (Ex. `/dir_1/data_2`, `/dir_1/data_3`, etc...).
+
+.Users will see a warning message in RESTPP log:
+[console]
+----
+[WARNING] The file "/dir_1/data_1" is skipped because it violates file input policy.
+----
+
+There are 2 ways to run loading jobs:
+
+1. The GSQL command:
++
+[console]
+----
+gsql -g G1 'run loading job load_job1'
+----
+2. The GSQL API:
++
+.Use the API to start the loading job and pass in the configuration json directly in string:
+[console]
+----
+curl --user tigergraph:tigergraph  -d '
+[
+   {
+      "name":"load_person",
+      "dataSources":[
+         {
+            "filename":"f1",
+            "name":"k1",
+            "path":"",
+            "config":{
+               "topic":"kiwi",
+               "partition_list":[
+                  {
+                     "start_offset":-2,
+                     "partition":0
+                  }
+               ]
+            }
+         }
+      ],
+      "streaming":false
+   }
+]
+' -X POST "http://localhost:8123/gsql/loading-jobs?graph=test_graph&action=start"
+----
+
+==== CREATE LOADING JOB
+
+If a path is explicitly given (Ex. `sys.data_root`) when creating a loading job, users can check the path during creation of the loading job and block it immediately if not allowed.
+
+== User-Created Query
+
+=== Installed Mode
+
+==== selectVertex
+
+`selectVertex` eill read existing vertices from a local file directly.
```

Comment:
typo?

Reply:
Indeed! :) I fixed this with last commit.